types/lib/sequelize.d.ts

```diff
@@ -166,7 +167,7 @@ export interface Config {
   };
 }
 
-export type Dialect = 'mysql' | 'postgres' | 'sqlite' | 'mariadb' | 'mssql';
+export type Dialect = 'mysql' | 'postgres' | 'sqlite' | 'mariadb' | 'mssql' | AbstractDialect;
```

Comment:
What we are gonna receive here is _not_ an _instance_ of `AbstractDialect`, that's why we can't pass it like this, because what we expect here is the class itself. If we wanna use the `AbstractDialect` type for that we can do it like this:

```suggestion
interface AbstractDialectConstructor {
  new(...args: any[]): AbstractDialect
}

export type Dialect = 'mysql' | 'postgres' | 'sqlite' | 'mariadb' | 'mssql' | AbstractDialectConstructor;
```